pytest_timeout.py

```diff
@@ -134,12 +134,34 @@ def pytest_enter_pdb():
     SUPPRESS_TIMEOUT = True
 
 
+def is_debugging():
+    """
+    Detects if a debugging session is in progress by checking if either of
+     the following conditions is true:
+        1. examining the stack frames to see if pydevd.py —
+     a debugging framework used by VSCode, PyCharm, and others — is present.
+        2. Check is SUPPRESS_TIMEOUT is set to True
+    """
+    global SUPPRESS_TIMEOUT
+    known_debugging_modules = {"pydevd"}
```

Comment:
Could you either make this a global constant or move it in the else clause?  I don't mind which.
I know it's probably a micro-optimisation of not having to construct the set and it's possibly even dubious that the global lookup would be faster than alwasy constructing such a tiny set but I think it's nicer style anyway.